x-pack/plugins/uptime/common/requests/get_certs_request_body.ts

```diff
@@ -0,0 +1,183 @@
+
+  const searchRequest = createEsQuery({
+    body: {
+      from: pageIndex * size,
+      size,
+      sort: asMutableArray([
+        {
+          [sort]: {
+            order: direction,
+          },
+        },
+      ]),
+      query: {
+        bool: {
+          ...(search
+            ? {
+                minimum_should_match: 1,
+                should: [
+                  {
+                    multi_match: {
+                      query: escape(search),
+                      type: 'phrase_prefix' as const,
+                      fields: [
+                        'monitor.id.text',
+                        'monitor.name.text',
+                        'url.full.text',
+                        'tls.server.x509.subject.common_name.text',
+                        'tls.server.x509.issuer.common_name.text',
+                      ],
+                    },
+                  },
+                ],
+              }
+            : {}),
+          filter: [
+            {
+              exists: {
+                field: 'tls.server.hash.sha256',
+              },
+            },
+            {
+              range: {
+                'monitor.timespan': {
+                  gte: from,
+                  lte: to,
+                },
+              },
+            },
+            ...(notValidBefore
+              ? [
+                  {
+                    range: {
+                      'tls.certificate_not_valid_before': {
+                        lte: notValidBefore,
+                      },
+                    },
+                  },
+                ]
+              : []),
+            ...(notValidAfter
+              ? [
+                  {
+                    range: {
+                      'tls.certificate_not_valid_after': {
+                        lte: notValidAfter,
+                      },
+                    },
+                  },
+                ]
+              : []),
+          ] as estypes.QueryDslQueryContainer,
+        },
+      },
+      _source: [
+        'monitor.id',
+        'monitor.name',
+        'tls.server.x509.issuer.common_name',
+        'tls.server.x509.subject.common_name',
+        'tls.server.hash.sha1',
+        'tls.server.hash.sha256',
+        'tls.server.x509.not_after',
+        'tls.server.x509.not_before',
+      ],
+      collapse: {
+        field: 'tls.server.hash.sha256',
+        inner_hits: {
+          _source: {
+            includes: ['monitor.id', 'monitor.name', 'url.full'],
+          },
+          collapse: {
+            field: 'monitor.id',
+          },
+          name: 'monitors',
+          sort: [{ 'monitor.id': 'asc' as const }],
+        },
+      },
+      aggs: {
+        total: {
+          cardinality: {
+            field: 'tls.server.hash.sha256',
+          },
+        },
+      },
+    },
+  });
+
+  return searchRequest.body;
+};
+
+export const processCertsResult = (result: CertificatesResults): CertResult => {
```

Comment:
```suggestion
export const processCertsResult = (result?: CertificatesResults): CertResult => {
```

We're optionally chaining `result` in several places, why not just mark it as an optional parameter to enforce that in the function. Or remove the optional chaining.